internal/session/handle_logout.go

```diff
@@ -14,6 +15,12 @@ func (s *Session) handleLogout(ctx context.Context, tag string, cmd *proto.Logou
 	s.capsLock.Lock()
 	defer s.capsLock.Unlock()
 
+	if s.state != nil {
```

Comment:
Already mentioned this before, but just for completeness: ideally, at least the way I understand RFC, there should be no logout-specific code, as clients can theoretically drop at any time. Any logout/cleanup code should be done in `Session.done` and things that it calls (e.g. `state.Close`).

Reply:
As commented in another comment this whole thing has been moved to `state.close()`